### ISSUE TITLE
Crash on Fire Emblem

### ISSUE THREAD

**Issue by ExPixel** [ExPixel/pyrite#8]
Crash during the intro of Fire Emblem. The crash doesn't occur if you skip it and the emulator moves onto the main menu just fine.

Trace:
```
executing: 0000034a eb00 [ ??? ]
r0 = 0x00000000;
r1 = 0x00000001;
r2 = 0x00000000;
r3 = 0x00000000;
r4 = 0x00000001;
r5 = 0x00000000;
r6 = 0x00000000;
r7 = 0x03007dd8;
r8 = 0x00000000;
r9 = 0x00000000;
r10 = 0x00000000;
r11 = 0x0000001f;
r12 = 0x04000000;
sp = 0x03007dc0;
	sp_irq = 0x03007fa0;
	sp_svc = 0x03007fd0;
	sp_fiq = 0x00000000;
	sp_abt = 0x00000000;
	sp_und = 0x00000000;
lr = 0x0000034c;
	lr_irq = 0x0000034c;
	lr_svc = 0x080bfa0e;
	lr_fiq = 0x00000000;
	lr_abt = 0x00000000;
	lr_und = 0x00000000;
pc = 0x0000034e
cpsr = 0x6000003f [ z c t ] [SYS]
spsr_irq = 0x6000003f
spsr_svc = 0x0000003f
spsr_fiq = 0x00000000
spsr_abt = 0x00000000
spsr_und = 0x00000000
========== BRANCHES ==========
0x0000013C (A) -> 0x00000348 (T) ; 1 times
0x03003A6C (A) -> 0x00000138 (A) ; 1 times
0x080BB4A2 (T) -> 0x03003A4C (A) ; 1 times
0x080BB482 (T) -> 0x080BB49E (T) ; 1 times
0x080BB46A (T) -> 0x080BB47A (T) ; 1 times
0x0000013C (A) -> 0x080BB468 (T) ; 1 times
0x03003A6C (A) -> 0x00000138 (A) ; 1 times
0x0801528C (T) -> 0x03003A4C (A) ; 1 times
0x080BE590 (T) -> 0x0801528A (T) ; 1 times
0x030027EE (T) -> 0x080BE58E (T) ; 1 times
0x030027D4 (T) -> 0x030027DA (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030027D8 (T) -> 0x030024E8 (T) ; 1 times
0x03002510 (T) -> 0x030027D0 (T) ; 1 times
0x030024F0 (T) -> 0x03002508 (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024DC (T) -> 0x030024CA (T) ; 1 times
0x030024C0 (T) -> 0x030024CA (T) ; 1 times
0x030024B8 (T) -> 0x030024BE (T) ; 1 times
0x03002454 (T) -> 0x030024B0 (T) ; 1 times
0x080BD8C8 (T) -> 0x03002450 (T) ; 1 times
0x080BD8B6 (T) -> 0x080BD8C2 (T) ; 1 times
0x080BF510 (T) -> 0x080BD8A8 (T) ; 1 times
0x080BF4FE (T) -> 0x080BF502 (T) ; 1 times
0x080BF10C (T) -> 0x080BF4F8 (T) ; 1 times
0x080BF500 (T) -> 0x080BF0F8 (T) ; 1 times
0x080BF10C (T) -> 0x080BF4F8 (T) ; 1 times
0x080BF500 (T) -> 0x080BF0F8 (T) ; 1 times
0x080BF10C (T) -> 0x080BF4F8 (T) ; 1 times
0x080BF500 (T) -> 0x080BF0F8 (T) ; 1 times
0x080BF10C (T) -> 0x080BF4F8 (T) ; 1 times
0x080BF0E6 (T) -> 0x080BF0F2 (T) ; 1 times
0x080BDC82 (T) -> 0x080BF0C8 (T) ; 1 times
0x080BD8A6 (T) -> 0x080BDC82 (T) ; 1 times
0x080BE158 (T) -> 0x080BD8A0 (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BE140 (T) -> 0x080BE148 (T) ; 1 times
0x080BE0AA (T) -> 0x080BE13E (T) ; 1 times
0x080BE146 (T) -> 0x080BE0A4 (T) ; 1 times
0x080BE0B0 (T) -> 0x080BE13E (T) ; 1 times
0x080BE09C (T) -> 0x080BE0A0 (T) ; 1 times
0x080BE088 (T) -> 0x080BE092 (T) ; 1 times
0x080BE076 (T) -> 0x080BE080 (T) ; 1 times
0x080BDF60 (T) -> 0x080BE074 (T) ; 1 times
0x080BE07E (T) -> 0x080BDF58 (T) ; 1 times
0x080BE028 (T) -> 0x080BE070 (T) ; 1 times
0x080BDF9C (T) -> 0x080BE014 (T) ; 1 times
0x080BDF6C (T) -> 0x080BDF96 (T) ; 1 times
0x080BDF5E (T) -> 0x080BDF62 (T) ; 1 times
0x080BE09E (T) -> 0x080BDF50 (T) ; 1 times
0x080BDF4E (T) -> 0x080BE098 (T) ; 1 times
0x080BDF44 (T) -> 0x080BDF48 (T) ; 1 times
0x080BEEB2 (T) -> 0x080BDF40 (T) ; 1 times
0x080BEDF4 (T) -> 0x080BEEAE (T) ; 1 times
0x080BDF3E (T) -> 0x080BEDEC (T) ; 1 times
0x080BDF30 (T) -> 0x080BDF34 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BDF32 (T) -> 0x080BE148 (T) ; 1 times
0x080BE158 (T) -> 0x080BDF1C (T) ; 1 times
0x080BE140 (T) -> 0x080BE148 (T) ; 1 times
0x080BE0C0 (T) -> 0x080BE134 (T) ; 1 times
0x080BEF66 (T) -> 0x080BE0BC (T) ; 1 times
0x080BEEC0 (T) -> 0x080BEF18 (T) ; 1 times
0x080BE0BA (T) -> 0x080BEEB4 (T) ; 1 times
0x080BE146 (T) -> 0x080BE0A4 (T) ; 1 times
0x080BE0C0 (T) -> 0x080BE134 (T) ; 1 times
0x080BEF66 (T) -> 0x080BE0BC (T) ; 1 times
0x080BEEC0 (T) -> 0x080BEF18 (T) ; 1 times
0x080BE0BA (T) -> 0x080BEEB4 (T) ; 1 times
0x080BE146 (T) -> 0x080BE0A4 (T) ; 1 times
0x080BE0C0 (T) -> 0x080BE134 (T) ; 1 times
0x080BEF66 (T) -> 0x080BE0BC (T) ; 1 times
0x080BEEC0 (T) -> 0x080BEF18 (T) ; 1 times
0x080BE0BA (T) -> 0x080BEEB4 (T) ; 1 times
0x080BE146 (T) -> 0x080BE0A4 (T) ; 1 times
0x080BE0C0 (T) -> 0x080BE134 (T) ; 1 times
0x080BEF66 (T) -> 0x080BE0BC (T) ; 1 times
0x080BEF00 (T) -> 0x080BEF04 (T) ; 1 times
0x080BEEF8 (T) -> 0x080BEEFE (T) ; 1 times
0x080BEEEA (T) -> 0x080BEEF2 (T) ; 1 times
0x080BEED0 (T) -> 0x080BEEDC (T) ; 1 times
0x080BE0BA (T) -> 0x080BEEB4 (T) ; 1 times
0x080BE146 (T) -> 0x080BE0A4 (T) ; 1 times
0x080BE0C0 (T) -> 0x080BE134 (T) ; 1 times
0x080BEF66 (T) -> 0x080BE0BC (T) ; 1 times
0x080BEEC0 (T) -> 0x080BEF18 (T) ; 1 times
0x080BE0BA (T) -> 0x080BEEB4 (T) ; 1 times
======== BRANCHES END ========
========== REGISTER CHANGES ==========
[0x00000138 (A)] r0 = 0x00000000 (~ previous 0x6000001f)
[0x00000138 (A)] r1 = 0x00000001 (~ previous 0x00002003)
[0x00000348 (T)] r2 = 0x00000000 (~ previous 0x00000000)
[0x00000138 (A)] r3 = 0x00000000 (~ previous 0x04000200)
[0x080BB49E (T)] r4 = 0x00000001 (~ previous 0x00000000)
[0x080BB49E (T)] r5 = 0x00000000 (~ previous 0x00000000)
[0x080BB49E (T)] r6 = 0x00000000 (~ previous 0x03001620)
[0x030027E2 (T)] r7 = 0x03007DD8 (~ previous 0x00000000)
[0x030027E4 (T)] r8 = 0x00000000 (~ previous 0x000000e0)
[0x030027E6 (T)] r9 = 0x00000000 (~ previous 0x00000000)
[0x030027E8 (T)] r10 = 0x00000000 (~ previous 0x00000000)
[0x030027EA (T)] r11 = 0x0000001F (~ previous 0x0000001f)
[0x00000138 (A)] r12 = 0x04000000 (~ previous 0x04000000)
[0x00000138 (A)] r13 = 0x03007FA0 (~ previous 0x03007f9c)
[0x00000138 (A)] r14 = 0x0000034C (~ previous 0x00000138)
[0x080BFA0C (T)] r15 = 0x00000008 (~ previous 0x080bfa10)
======== REGISTER CHANGES END ========
thread '<main>' panicked at 'picnic -- undefined instruction', src/gba/core/cpu/mod.rs:531
note: Run with `RUST_BACKTRACE=1` for a backtrace.
error: Process didn't exit successfully: `target/release/pyrite roms/legal/fire-emblem.gba` (exit code: 101)
```

**Comment by Charile1** [comment]
Better put that in high-priority, we have to make sure FireRed AND Fire Emblem works

-Charile

**Comment by ExPixel** [comment]
Fixed by a39efe804c2723384661ccb26c9e2642d6d0df39
